### PR TITLE
Add prometheus metrics for Central controllers

### PR DESCRIPTION
**Pull request by glimberg**

### CODE REVIEW

**Comment by rcoder** [comment]
I think this is a great start, and don't have any significant suggestions for structure or usage.

I've added a couple of comments around naming + where we put metric definitions in the code, but I'm eager to see this land so we can start doing similar instrumentation of the rest of our services. :)

**Comment by glimberg** [comment]
Where did you make comments?  I'm not seeing anything

File: controller/DB.hpp [hunk 1]
```diff
@@ -191,8 +191,12 @@ class DB
 	mutable std::mutex _changeListeners_l;
 	mutable std::mutex _networks_l;
 
-	prometheus::simpleapi::counter_metric_t _network_count { "contrller_network_count", "number of networks the controller is serving" };
-	prometheus::simpleapi::counter_metric_t _member_count { "controller_member_count", "number of network members the controller is serving" };
+	prometheus::simpleapi::gauge_metric_t _network_count { "controller_network_count", "number of networks the controller is serving" };
```
[the remark this answers is not in the document]
Comment (rcoder) [reply controller/DB.hpp]: I think you addressed this in your latest commit.

**Comment by rcoder** [comment]
Sorry for the dual reviews -- your latest commit landed while my comments were "pending".

Still looking good. :)

**Comment by Boilerplate4u** [comment]
@glimberg, just curious but why not just add the counters directly to the pool methods like push_back, pop_front and so on? Class init might control whether metrics should be used or not...

File: controller/DB.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (glimberg) [reply controller/DB.hpp]: I did attempt one method, by consolidating all the metrics into a single place, however things appear to explode, unfortunately.

File: controller/ConnectionPool.hpp [hunk 2]
```diff
@@ -99,6 +103,7 @@ class ConnectionPool {
                 try {
                     std::shared_ptr<Connection> conn = m_factory->create();
                     m_borrowed.insert(conn);
+                    _pool_in_use++;
                     return std::static_pointer_cast<T>(conn);
                 } catch (std::exception &e) {
                     throw ConnectionUnavailable();
```
[the remark this answers is not in the document]
Comment (glimberg) [reply controller/ConnectionPool.hpp:110]: done

**Pull request merged**